src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/DebugPageControl.xaml.cs

```diff
@@ -52,17 +52,17 @@ private void OnFocusEnvironmentVariableGridRow(object sender, EventArgs e)
         {
             if (DataContext != null && DataContext is DebugPageViewModel)
             {
-                Dispatcher.BeginInvoke(new DispatcherOperationCallback((param) =>
+                ThreadHelper.JoinableTaskFactory.StartOnIdle(async () =>
                 {
+                    await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
                     if ((DataContext as DebugPageViewModel).EnvironmentVariables.Count > 0)
                     {
                         // get the new cell, set focus, then open for edit
                         var cell = WpfHelper.GetCell(dataGridEnvironmentVariables, (DataContext as DebugPageViewModel).EnvironmentVariables.Count - 1, 0);
                         cell.Focus();
                         dataGridEnvironmentVariables.BeginEdit();
                     }
-                    return null;
-                }), DispatcherPriority.Background, new object[] { null });
+                });
```

Comment:
As @jmarolf taught me today that `Dispatcher.BeginInvoke` delegates would crash VS, I'll call out that `StartOnIdle` returns a faulting `JoinableTask` instead of crashing. So if you want to notice any failures in this delegate, consider adding `.FileAndForget("vs/your/telemetry/event")` just before the semicolon on this line.